[kararei] iP

kararei opened the pull request:
### kx (kaixin)

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

kaixin frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from ([here](https://github.com/kararei/ip/releases))
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main `method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kararei committed:
add AI.md AI records

kararei committed:
Recreated to kx with greeting

kararei committed:
Add echo ability

kararei committed:
Add listing and adding ability

kararei committed:
Add Mark as Done Commands

kararei committed:
Add Deadlines, Events, Todo files

kararei committed:
Add event, todo, deadline handlers

kararei committed:
Setup automated Text UI testing

kararei committed:
Add error handling

kararei committed:
Add task deletion support

kararei committed:
Fix formatting errors

kararei committed:
Add dataStore exist check

isaacchua0309 commented on src/main/java/kx.java:
```diff
@@ -1,5 +1,10 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
```
// ⚠️ Duplicate import of java.io.IOException. Consider removing one to avoid redundancy.

isaacchua0309 commented on src/main/java/kx.java:
```diff
@@ -1,5 +1,10 @@
+import java.io.FileWriter;
+import java.io.IOException;
 
 public class kx {
```
// ⚠️ Class name should follow PascalCase (e.g., Kx) according to Java naming conventions.

isaacchua0309 commented on src/main/java/kx.java:
```diff
@@ -109,12 +114,29 @@ private static void listOut(ArrayList<Task> tasks) {
         System.out.println(SEPARATOR);
```
// ⚠️ Consider using a loop to iterate over tasks and print them individually for better clarity.

isaacchua0309 commented on src/main/java/kx.java:
```diff
@@ -109,12 +114,29 @@ private static void listOut(ArrayList<Task> tasks) {
         tasks.add(newTask);
             System.out.println(SEPARATOR);
             System.out.println("  Got it. I've added this task:");
             System.out.println("  " + newTask.toString());
```
// ⚠️ newTask.toString() is redundant; println() calls toString() automatically.

isaacchua0309 reviewed (commented):
Isaac Chua (isaacchua0309) PR Review on kararei

lilyium commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadlines extends Task{
```
Should the class name be changed to a singular noun to make it conistent?

lilyium commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,16 @@
+public class Events extends Task {
```
Should the class name be changed to a singular noun to make it conistent?

lilyium commented on src/main/java/kx.java:
You might want to consider changing the class name to begin with a capital letter.

lilyium commented on src/main/java/kxException.java:
```diff
@@ -0,0 +1,6 @@
+public class kxException extends Exception {
```
You might want to consider changing the class name to begin with a capital letter.

lilyium commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    public String markAsDone() {
+        this.isDone = true;
+        return kx.SEPARATOR + "\n  Nice! I've marked this task as done:\n  " + this.toString() + "\n" + kx.SEPARATOR;
+    }
+
+
+    public String markAsUndone() {
+        this.isDone = false;
+        return kx.SEPARATOR + "\n  OK, I've marked this task as not done yet:\n  " + this.toString() + "\n" + kx.SEPARATOR;
+    }
```
Good method naming, it is easy to tell the intent of the method.

lilyium reviewed (commented):
LGTM👍!

kararei committed:
Update data storage ability

kararei committed:
Add DateTime formatting ability

kararei committed:
Merge branch 'branch-Level-8'
Merge Level-8 branch into master

kararei committed:
Refactor files with more OOP

kararei committed:
Merge branch 'MoreOOP'
Refactor code for more OOP

kararei committed:
Package classes into subsets

kararei committed:
Merge branch 'add-gradle-support'
Add gradle support

kararei committed:
Edit gradle build file

kararei committed:
Add JUnit Test for Parser and Ui class

kararei committed:
Configure gradle for jar file

kararei committed:
Add JavaDoc

kararei committed:
Edit code following coding standards

kararei committed:
Add find matching tasks ability

kararei committed:
"Merge branch-A-CodingStandards with master"

kararei committed:
"Merge branch-A-CodingStandards with master"

kararei committed:
Add JavaFX GUI to kx chatbot

This commit integrates a graphical user interface (GUI) into the chatbot using JavaFX, replacing the previous text-based UI. Users can now interact with the chatbot via a graphical interface instead of the command line.

- Implemented JavaFX-based GUI following the JavaFX tutorial from @SE-EDU/guides.
- Completed up to part 4 of the tutorial before integrating JavaFX into the chatbot.
- Created a separate `Launcher` class to handle classpath issues.
- Designed a functional UI with `MainWindow.fxml` to enable message input and display chatbot responses.
- Replaced `Scanner`-based user input with JavaFX event handling.
- Updated `kx`, `Parser`, `Ui`, and `Storage` classes to work with JavaFX.
- Removed the text-based UI and outdated I/O redirection-based testing (A-TextUiTesting).

kararei committed:
Add delayed exit on 'bye' and auto 'hello' message

This commit enhances the kx chatbot user experience through
1. Displaying a greeting message automatically when the application
   starts
2. Ensuring the 'bye' command shows an exit message and waits for a few
   seconds before the application closes

This is to improve the UX
Users can still close the java window manually

kararei committed:
Merge Branch-Level-10: Add javafx gui

kararei committed:
Add assert feature to document assumptions

kararei committed:
Merge pull request #2 from kararei/branch-A-Assertions

Add assert feature to document assumptions

kararei committed:
Improve code quality

kararei committed:
Refactor parser into smaller methods

kararei committed:
"Merge master with branch-A-CodeQuality"

kararei committed:
Merge pull request #3 from kararei/branch-A-CodeQuality

Branch a code quality

kararei committed:
Add B-ViewSchedules extension

kararei committed:
Update README description

kararei committed:
Edit README and add title

kararei committed:
Update Ui testing file

kararei committed:
Improve exception handling

kararei committed:
Add css files

kararei committed:
Update JavaDoc comments in Parser

kararei committed:
Update mark and unmark excpetion handling

kararei committed:
Handle wrong DateTime format exceptions
